packages/background/src/controllers/BlankProviderController.ts

```diff
@@ -1085,8 +1087,15 @@ export default class BlankProviderController extends BaseController<BlankProvide
             intervalRef && clearInterval(intervalRef);
             timeoutRef && clearTimeout(timeoutRef);
         };
+        const __clearListeners = () => {
+            this.removeAllListeners(
+                KeyringControllerEvents.QR_MESSAGE_SIGNATURE_REQUEST_GENERATED
+            );
+        };
 
         try {
+            __clearListeners();
```

Comment:
I'm wondering if we should invoke this `__clearListeners` at the end of this function too.